### PR TITLE
Update status checks for Gold Status -> Github

### PR DESCRIPTION
**Pull request by matanlurey**
These checks are out of date:

- It's `web_engine` not `web engine`
- We're missing Impeller, Scenario App goldens for the native engine entirely

/cc @mdebbar as well.

I also filed https://github.com/flutter/flutter/issues/143863 since it's likely we'll drift again in the future.

### CODE REVIEW

File: app_dart/lib/src/request_handlers/push_gold_status_to_github.dart [hunk 1]
```diff
@@ -119,11 +119,22 @@ class PushGoldStatusToGithub extends ApiRequestHandler<Body> {
       for (Map<String, dynamic> checkRun in checkRuns) {
         log.fine('Check run: $checkRun');
         final String name = checkRun['name'].toLowerCase() as String;
-        // Framework shards run framework goldens
-        // Web shards run web version of framework goldens
-        // Misc shard runs API docs goldens
-        if (name.contains('framework') || name.contains('web engine') || name.contains('misc')) {
-          runsGoldenFileTests = true;
+        if (slug == Config.engineSlug) {
+          if (const <String>[
```
Comment (jonahwilliams) [inline app_dart/lib/src/request_handlers/push_gold_status_to_github.dart:123]: nit, this can be a Set :)

**Review by jonahwilliams — APPROVED** [review]
LGTM

File: app_dart/lib/src/request_handlers/push_gold_status_to_github.dart (hunk 1 above)
Comment (matanlurey) [reply app_dart/lib/src/request_handlers/push_gold_status_to_github.dart:123]: Oh wait I actually broke this, it needs to be looser

**Review by mdebbar — APPROVED** [review]
LGTM thanks for updating this!

File: app_dart/lib/src/request_handlers/push_gold_status_to_github.dart (hunk 1 above)
Comment (matanlurey) [reply app_dart/lib/src/request_handlers/push_gold_status_to_github.dart:123]: Fixed.

**Review by mdebbar — CHANGES_REQUESTED** [review]

File: app_dart/lib/src/request_handlers/push_gold_status_to_github.dart [hunk 2]
```diff
@@ -119,11 +119,22 @@ class PushGoldStatusToGithub extends ApiRequestHandler<Body> {
...
+            'linux_android_emulator',
+            'linux_host_engine',
+            'mac_host_engine',
+            'linux_web_engine',
+          ].any((String shard) => shard.contains(name))) {
```
[the remark this answers is not in the document]
Comment (mdebbar) [reply app_dart/lib/src/request_handlers/push_gold_status_to_github.dart:128]: No, I'm looking at your last commit that "Fixes the contains check" :)

The condition should be `name.contains(...one of the hardcoded values...)` not the other way around.

[the remark this answers is not in the document]
Comment (matanlurey) [reply app_dart/lib/src/request_handlers/push_gold_status_to_github.dart:128]: Oh rip, you're right. BUT THE UNIT TESTS SAID IT WAS FINE :P

[the remark this answers is not in the document]
Comment (mdebbar) [reply app_dart/lib/src/request_handlers/push_gold_status_to_github.dart:128]: You want to check if the check name contains the word "framework".
You don't want to check if the word "framework" contains the check name.

[the remark this answers is not in the document]
Comment (mdebbar) [reply app_dart/lib/src/request_handlers/push_gold_status_to_github.dart:128]: > Oh rip, you're right. BUT THE UNIT TESTS SAID IT WAS FINE :P

Yeah, we should update the tests to something more realistic e.g. "Linux linux_web_engine"

[the remark this answers is not in the document]
Comment (matanlurey) [reply app_dart/lib/src/request_handlers/push_gold_status_to_github.dart:128]: Done

File: app_dart/test/request_handlers/push_gold_status_to_github_test.dart [hunk 3]
```diff
@@ -767,7 +767,7 @@ void main() {
             <String, String>{'name': 'framework', 'status': 'in_progress', 'conclusion': 'neutral'},
           ];
           engineCheckRuns = <dynamic>[
-            <String, String>{'name': 'web engine', 'status': 'in_progress', 'conclusion': 'neutral'},
+            <String, String>{'name': 'linux_web_engine', 'status': 'in_progress', 'conclusion': 'neutral'},
```
Comment (mdebbar) [inline app_dart/test/request_handlers/push_gold_status_to_github_test.dart]:
```suggestion
            <String, String>{'name': 'Linux linux_web_engine', 'status': 'in_progress', 'conclusion': 'neutral'},
```

File: app_dart/test/request_handlers/push_gold_status_to_github_test.dart [hunk 4]
```diff
@@ -1482,7 +1482,7 @@ void main() {
           <String, String>{'name': 'framework', 'status': 'completed', 'conclusion': 'success'},
         ];
         engineCheckRuns = <dynamic>[
-          <String, String>{'name': 'web engine', 'status': 'completed', 'conclusion': 'success'},
+          <String, String>{'name': 'linux_web_engine', 'status': 'completed', 'conclusion': 'success'},
```
Comment (mdebbar) [inline app_dart/test/request_handlers/push_gold_status_to_github_test.dart]:
```suggestion
          <String, String>{'name': 'Linux linux_web_engine', 'status': 'completed', 'conclusion': 'success'},
```

**Review by mdebbar — APPROVED** [review]

**Review by ricardoamador — APPROVED** [review]
LGTM

**Review by Piinks — APPROVED** [review]
LGTM

File: app_dart/lib/src/request_handlers/push_gold_status_to_github.dart [hunk 5]
```diff
@@ -119,11 +119,22 @@ class PushGoldStatusToGithub extends ApiRequestHandler<Body> {
...
+            runsGoldenFileTests = true;
+          }
+        } else if (slug == Config.flutterSlug) {
+          if (const <String>[
+            'framework',
+            'misc',
+          ].any((String shard) => shard.contains(name))) {
+            runsGoldenFileTests = true;
```
Comment (Piinks) [inline app_dart/lib/src/request_handlers/push_gold_status_to_github.dart]: Just marking to cross link.

**Pull request merged**